azext_edge/tests/edge/orchestration/resources/test_instance_int.py

```diff
@@ -13,6 +13,9 @@
 
 logger = get_logger(__name__)
 
+# pytest mark for rpsaas (cloud-side) tests
+pytestmark = pytest.mark.rpsaas
+
```

Comment:
if I add in a kubectl see that the assets are there after create - would this be considered a e2e or still rpsaas?

Reply:
pretend I commented in the asset int test file

Reply:
also instance does have cluster side checks

Reply:
> if I add in a kubectl see that the assets are there after create - would this be considered a e2e or still rpsaas?

That is an interesting situation, though I think it's probably fine to mark as `rpsaas`. An `rpsaas` test can technically also be marked as `e2e`, but the important distinctions are:

- Tests marked `e2e` are designated to be run in the AIO pipeline, and we want to be careful how we introduce tests with cloud side dependencies to these, since these may interfere / interact with the other AIO pipeline stuff. Namely (at least currently) - we don't want to block PRs for service flakiness, and we don't want to do anything instance/update/init related to the cluster since other tests components from the E2E tests rely on the cluster instance.

- Tests marked `rpsaas` are designated as having a cloud-side dependency. They can be run in other groupings like our normal integration tests, but they have this mark since they are typically longer running and highly dependent on an RP. This is *mostly* for local testing - so you can run the `edge` tox scenario to run quicker tests that only tests cluster-side stuff, or the `rpsaas` tests if you're mostly concerned about the cloud interactions.

A test can of course have both marks, but giving it the `rpsaas` mark will exclude it from the tox `edge` environment, and giving it the `e2e` mark will include it in external tests.

I will add this logic and explanation to the testing document with some other updates to reflect this PR's changes